### ISSUE TITLE
Contracts on builtins

### ISSUE THREAD

**Issue by jpolitz** [brownplt/pyret-lang#48]
Many builltins, like `_divide` and `substring`, are partial, and give contract errors for erroneous inputs.  We should turn these errors into meaningful Pyret errors somehow, and avoid paying to much to do so (that's the hard part).

**Comment by schanzer** [comment]
_divide(1,"2") seems to give Pyret-esque errors now, no?

**Comment by blerner** [comment]
When I try it, I get an internal error (field not found "left" on an s-program).  @jswrenn over to you...

**Comment by schanzer** [comment]
@jpolitz is this still an issue?

**Comment by blerner** [comment]
Yes, it's still exposing an internal error.

**Comment by schanzer** [comment]
In 2026, we get a slightly-better error:

<img width="713" height="378" alt="Image" src="https://github.com/user-attachments/assets/e72590e3-d709-4152-8407-2064bcecea2c" />

**Comment by blerner** [comment]
No, that's exactly the same internal error that was exposed 6 years ago...

**Comment by schanzer** [comment]
Ah, I thought the internal error it was giving back then was the old "An internal error prevented this from being rendered" blurb.
